test/unit/request-auth.test.js

```diff
@@ -123,4 +123,85 @@ describe('RequestAuth', function () {
             fakeParent: true
         });
     });
+
+    describe('.update', function () {
```

Comment:
If I have to follow TDD then I would like this test suite to be:

```
#update
- should update the parameters of a specific authentication type
- should update the parameters of the existing type if not specified explicitly
- should bail out if invalid auth type is provided
```

Reply:
Fixed.